blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubPipelineCreateRequest.java

```diff
@@ -108,21 +128,53 @@ public BluePipeline create(Reachable parent) throws IOException {
                                         ));
                     }
                 }
-                GitHubSCMNavigator gitHubSCMNavigator = new GitHubSCMNavigator(apiUrl, orgName, credentialId, credentialId);
-                if (sb.length() > 0) {
-                    gitHubSCMNavigator.setPattern(sb.toString());
-                }
 
                 // cick of github scan build
                 OrganizationFolder organizationFolder = (OrganizationFolder) item;
+
+                GitHubSCMNavigator gitHubSCMNavigator = organizationFolder.getNavigators().get(GitHubSCMNavigator.class);
+
+                StringBuilder sb = new StringBuilder();
+                if (gitHubSCMNavigator != null) {
+                    Matcher matcher = Pattern.compile("\\((.*?)\\\\b\\)\\?").matcher(gitHubSCMNavigator.getPattern());
+
+                    while (matcher.find()) {
+                        String existingRepo = matcher.group(1);
+                        if (!repos.contains(existingRepo)) {
+                            repos.add(existingRepo);
+                        }
+                    }
+                }
+
+                gitHubSCMNavigator = new GitHubSCMNavigator(apiUrl, orgName, credentialId, credentialId);
                 organizationFolder.getNavigators().replace(gitHubSCMNavigator);
 
-                if(repos.size() == 1){
-                    SCMSourceEvent.fireNow(new SCMSourceEventImpl(repos.get(0), item, apiUrl, gitHubSCMNavigator));
+                for (String r : repos) {
+                    sb.append(String.format("(%s\\b)?", r));
+                }
+                if (sb.length() > 0) {
+                    gitHubSCMNavigator.setPattern(sb.toString());
+                }
```

Comment:
In general maybe some of this regexp handling code would benefit from method extraction but that's just a style thing, no biggie.